src/kremling-inline-loader.utils.js

```diff
@@ -0,0 +1,136 @@
+exports.testKremling = function(src, i, config) {
```

Comment:
Damn. this is an entire javascript parser. It seems pretty likely to be incomplete/inaccurate, especially when considering things like jsx, typescript, flow, and other random babel plugins. The grammar for ecmascript is quite complicated, and i don't think we'll be able to recreate it.

The current complexity of this function, along with the inevitability of bugs and future complexity, makes me nervous about this implementation approach.

An alternative that I would prefer is to make this a babel plugin instead of a webpack loader. This would require usage of babel for all users, but would guarantee that the javascript gets parsed correctly and we don't have to maintain our own parser. The [emotion library does this](https://emotion.sh/docs/css-prop#babel-preset) for their implementation of the `css` prop, and i think it would make me feel a lot better about the implementation if we did something similar.

Writing a babel plugin is something i've done before, so I can give you some tips if you'd like (hmu on single-spa slack workspace). But i honestly think that it's the only way to make the implementation of `k` bug free and maintainable.

Reply:
yeah - it was a bit naive of me to take this path... but I figured it was worth a shot 🤷‍♂️ (and it's been fun writing it)

Concerning your previous comment, it's actually the reason why I didn't create another npm package - it was a first-pass at an idea... something we could dump if it didn't work out (I think the kremling `k` tag is the real winner here)

I did go down the babel-plugin idea, but there are 2 main issues with it:

- template literals contain placeholders `${}` that are not parsable by postcss (I have some ideas on how to handle this...)
- MAIN ISSUE: some postcss plugins (like cssnano) are async-only - babel is completely synchronous

we could get around the async issue by not allowing some postcss plugins... but that sounds kinda lame